include/zephyr/net/zperf.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2016 Intel Corporation
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file zperf.h
+ *
+ * @brief Zperf API
+ * @defgroup zperf Zperf API
+ * @ingroup networking
+ * @{
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_ZPERF_H_
+#define ZEPHYR_INCLUDE_NET_ZPERF_H_
+
+#include <zephyr/net/net_ip.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum zperf_status {
+	ZPERF_SESSION_STARTED,
+	ZPERF_SESSION_FINISHED,
+	ZPERF_SESSION_ERROR
+} __packed;
+
+struct zperf_upload_params {
+	struct sockaddr peer_addr;
+	uint32_t duration_ms;
+	uint32_t rate_kbps;
+	uint16_t packet_size;
+	struct {
+		uint8_t tos;
+	} options;
+};
+
+struct zperf_results {
+	uint32_t nb_packets_sent;
+	uint32_t nb_packets_rcvd;
+	uint32_t nb_packets_lost;
+	uint32_t nb_packets_outorder;
+	uint32_t total_len;
+	uint32_t time_in_us;
+	uint32_t jitter_in_us;
+	uint32_t client_time_in_us;
+	uint32_t packet_size;
+	uint32_t nb_packets_errors;
+};
+
+/**
+ * @brief Zperf callback function used for asynchronous operations.
+ *
+ * @param status Session status.
+ * @param result Session results. May be NULL for certain events.
+ */
+typedef void (*zperf_callback)(enum zperf_status status,
+			       struct zperf_results *result);
+
```

Comment:
Perhaps this could be placed to the commit that introduces the async API?

Reply:
True